[3.3] Separated change monitoring for exposure counts from other changes to a sequence.  Allow users to ignore exposure count changes.  General HasChanges clean up

ste7e opened the pull request:
<!--
🎯 Thank you for contributing to N.I.N.A.! Please fill out the template below to help us review your PR effectively.
-->

## 🚀 Purpose

<!-- Briefly describe the goal of this pull request. What feature, fix, or improvement does it bring? -->

Whenever a sequence takes an exposure, the exposure count changes.  At the moment, this is considered as a change to the sequnce so the user gets prompted to save the sequence after every run.  They don't know whether that's because they've made a meaningful change the sequence or if it's just that the exposure count has changed.

This change separates out the exposure count from other sequence items.  This is done by decorating the exposure count property with a new attribute (HasChangedSetAttribute) that collects changes to it in a different HasChanges flag to the non-decorated properties.

Example:         
        `[HasChangedSet("Exposures")]`
 

Now, if the user doesn't change the sequence but does run it, they'll still be prompted to save the sequence - to retain the exposure count - but the message will tell them it's the exposure count that's changed so they can decide whether to save it or not.

In addition, there is now a new option in the Sequences section, to allow the user to suppress the prompts about changes to exposure count - by using this they can chose to ignore changes to exposure count.  This means that they are not prompted to save the sequence after every run.

## 🧪 How Was It Tested?

<!-- Describe how you tested your changes. Include manual tests, automated tests, and testing across themes or configurations. -->
Manual tests - making changes to sequences - advanced and simple - and checking what prompts for a save and what doesn't

## ✅ PR Checklist

- [ ] All unit tests pass
- [ ] UI changes tested across Light, Dark, and Night themes (if applicable)
- [ ] Plugin API compatibility reviewed  
  - [ ] No breaking changes to interfaces  
  - [ ] No changes to method/class signatures (especially optional parameters)
- [ ] `Changelog.md` updated (if applicable)
- [ ] [Documentation](https://github.com/isbeorn/nina.docs) updated (if applicable)

## 🔗 Related Issues

<!-- List related GitHub issues this PR closes or is connected to. Use GitHub keywords (e.g., "Closes #123"). -->

## 📸 Screenshots

<!-- If your PR includes UI changes, include before/after screenshots or videos -->

## 📝 Additional Notes

<!-- Add any extra context, caveats, or considerations for reviewers here -->

ste7e committed:
Working but not as slick as I'd like

ste7e committed:
HasChanges flags moved up to SequenceRootContainer for advanced sequencer and SimpleDSOContainer for simple.  These are set and checked in the VMs (Sequence2VM & SimpleSequenceVM).  Also set via INPC in SequenceHasChanged class (base class of sequence items)

ste7e committed:
Removed unnecessary ISequenceHasChanges and renamed SequenceHasChanges to more meaningful SequenceEntityINPC

ste7e committed:
Separated change monitoring for exposure counts from other changes
to a sequence.  Allows user to distinguish a modified sequence which needs saving
from a sequence which has taken exposures and so has a new exposure
count.  The user can then chose to save the exposure count or not.

Added option to allow user to ignore prompts to save exposure count.

Generally tidies up and improved the HasChanges flag.  HasChanges
flags moved up to SequenceRootContainer for advanced
sequencer and SimpleDSOContainer for simple instead of holding the
flag aginst eact sequence item.  These are set and checked
in the VMs (Sequence2VM & SimpleSequenceVM).  Also set via INPC in
SequenceHasChanged class (base class of sequence items)

I've made changes to Locale.resx and removed the old version of the
changed entry from Locale.en-GB.resx.  Not sure if that's the correct
process for the Locale resx files.  Should I remove the old string from
all the resx files to prompt a new translation of the new version?

isbeorn commented on NINA.Core/Locale/Locale.en-GB.resx:
```diff
@@ -575,11 +575,6 @@ able to compensate for ambient temperature variations</value>
   <data name="LblChange" xml:space="preserve">
     <value>Change</value>
   </data>
-  <data name="LblChangedSequenceWarning" xml:space="preserve">
```
please only change locale.resx. the other locale files will be updated once they have run through crowdin.

ste7e committed:
Removed unchanged solution file and Locale.en-GB.resx

isbeorn commented on NINA.Sequencer/SequenceEntityINPC.cs:
```diff
@@ -0,0 +1,76 @@
+            //Trace.WriteLine(_profileService.ActiveProfile.ImageFileSettings.FilePath);
+
+            System.Reflection.PropertyInfo propInf = GetType().GetProperty(e.PropertyName);
+            if (propInf.GetCustomAttributes(typeof(JsonPropertyAttribute), true).Length > 0) {
```
it might be a good idea to cache this in a map of PropertyName -> PropertyMeta, to only have the reflection cost once

ste7e commented on NINA.Sequencer/SequenceEntityINPC.cs:
```diff
@@ -0,0 +1,76 @@
+        }
+
+
+        public virtual ISequenceRootContainer GetSequenceRootContainer() {
```
OK, I've changed this method now so that it uses the ItemUtility method

ste7e committed:
GetSequenceRootContainer now makes use of ItemUtility.GetRootContainer
PropertyInfo is now cached in SequenceEntityINPC.OnPropertyChanged
defaultChangedSet constant added ("*")

isbeorn committed:
Merge branch 'develop' into develop

isbeorn committed:
Merge branch 'develop' into develop

isbeorn committed:
Add missing data closing tag

isbeorn reviewed (changes requested):
SequenceContainerTest needs to be updated with the new logic

ste7e commented:
> SequenceContainerTest needs to be updated with the new logic

Will try to get this done today

ste7e committed:
Created helper GetRootContainer() as GetRootContainer(this)
Changed SequenceContainerTest for new simplified logic

ste7e committed:
Merge branch 'develop' of github.com:ste7e/Nina into develop

isbeorn committed:
Merge branch 'develop' into develop

isbeorn committed:
Merge branch 'develop' into develop

isbeorn committed:
Revert solution guid changes
